mdp/utils/covariance.py

```diff
@@ -177,33 +177,46 @@ def update(self, x, dt=None):
             and random variables along the columns.
         :type x: numpy.ndarray
 
-        :param dt: Sequence of time increments between vectors. Must be
-            of length *x.shape[0]-1* in case of a single call. In case of
-            multiple calls with intended time dependence - as opposed to
-            computing the results as a weighted avarage of separate calls -
-            *dt* must be of length *x.shape[0]* starting with the second call.
-            The first element will be considered as the time difference
-            between the last element of *x* in the preceding call and the
-            first element of *x* in the current call. In case of not supplying
-            a *x.shape[0]*-th element in *dt*, time dependence will be dropped.
-            If *dt* omitted entirely, it will be considered to be one everywhere
-            whereas time dependence between calls is still dropped.
+        :param dt: Sequence of time increments between vectors. 
+
+            Usage with only single chunk of data:
+                *dt* must be of length *x.shape[0]-1*.
+
+            Usage with multiple chunks of data with intended time dependence:
+                *dt* should have length *x.shape[0]-1* in the first call and
```

Comment:
Please make this less strict (also in the case of just a single chunk). Allow `len(dt) == x.shape[0]` also in first call and ignore the additional element. Raise a warning then. This gives the user more flexibility in handling, e.g. allows to reuse the same array for data buffering without case distinction for the first call and additional slicing.

Reply:
Okay.